src/app/components/shared/delete-modal/delete-modal.component.ts

```diff
@@ -0,0 +1,49 @@
+import { Component, Input } from "@angular/core";
+import { PageComponent } from "@helpers/page/pageComponent";
+import { withUnsubscribe } from "@helpers/unsubscribe/unsubscribe";
+import { ModalComponent } from "@menu/widget.component";
+import { SharedActivatedRouteService } from "@services/shared-activated-route/shared-activated-route.service";
+
+@Component({
+  selector: "baw-delete-modal",
+  template: `
+    <div class="modal-header">
+      <h4 class="modal-title">Delete Item</h4>
+      <button
+        type="button"
+        class="btn-close"
+        aria-label="Close"
+        (click)="dismiss()"
+      ></button>
+    </div>
+
+    <div class="modal-body">
+        <span id="subTitle">
+          <p>Are you certain you wish to delete this item?</p>
+        </span>
+    </div>
+
+    <div class="modal-footer">
+      <a class="btn btn-outline-primary" (click)="dismiss()">Cancel</a>
+      <a class="btn btn-danger text-white" (click)="deleteModel()">Delete</a>
+    </div>
+  `,
+})
+export class DeleteModalComponent extends withUnsubscribe() implements ModalComponent {
+  public constructor(
+    private sharedRoute: SharedActivatedRouteService,
+  ) { super(); }
+
+  @Input() public successCallback: (component: PageComponent) => void;
+  public closeModal: (result: any) => void;
+  public dismissModal: (reason: any) => void;
+
+  protected deleteModel(): void {
+    this.successCallback(this.sharedRoute.pageComponentInstance);
```

Comment:
you just need to invoke success callback, shared route should be invoking the menuitem callback with the parameter

delete modals shouldn't know about shared menus.

if they don't know then we could put the delete button in the page itself, or in the menu widget list. either works!

Reply:
I have implemented this through the [menu service](https://github.com/QutEcoacoustics/workbench-client/pull/2042/files#diff-ab6a0ee0e398612e0fbfa14ca21cf089c65fb17ad1335d99e1069a8f583616d4R165-R173) that adds the component information to the callback.

+ 2 lines of additional code to the [menu components](https://github.com/QutEcoacoustics/workbench-client/pull/2042/files#diff-c9315700578bd5e26c449a32c33915231c2c45e8e4246f2767f46491f448eaafR136-R138) `createModalAction` method.